plugin/ochttp/trace.go

```diff
@@ -53,11 +56,80 @@ func (t *traceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	name := spanNameFromURL(req.URL)
 	// TODO(jbd): Discuss whether we want to prefix
 	// outgoing requests with Sent.
-	_, span := trace.StartSpan(req.Context(), name,
+	ctx, span := trace.StartSpan(req.Context(), name,
 		trace.WithSampler(t.startOptions.Sampler),
 		trace.WithSpanKind(trace.SpanKindClient))
 
-	req = req.WithContext(trace.WithSpan(req.Context(), span))
+	tracer := &httptrace.ClientTrace{
+		GetConn: func(hostPort string) {
+			gctx, span := trace.StartSpan(ctx, "GetConn")
+			ctx = context.WithValue(gctx, "getConnSpan", span)
+		},
+		GotConn: func(_ httptrace.GotConnInfo) {
+			span := ctx.Value("getConnSpan").(*trace.Span)
+			span.End()
+		},
+		DNSStart: func(_ httptrace.DNSStartInfo) {
+			dctx, span := trace.StartSpan(ctx, "DNSLookup")
+			ctx = context.WithValue(dctx, "dnsSpan", span)
+		},
+		DNSDone: func(di httptrace.DNSDoneInfo) {
+			span := ctx.Value("dnsSpan").(*trace.Span)
+			if di.Err != nil {
+				span.SetStatus(trace.Status{
+					Code:    trace.StatusCodeInternal,
+					Message: di.Err.Error(),
+				})
+			}
+			span.End()
+		},
+		WroteHeaders: func() {
+			span := trace.FromContext(ctx)
+			span.Annotate(nil, "Wrote request")
```

Comment:
Annotation should be "wrote request headers" - might be misleading to just say "wrote request"

Reply:
Gotcha, thanks!